app/src/main/java/com/lpnu/mobile/fragments/Details.java

```diff
@@ -0,0 +1,137 @@
+    @BindView(R.id.add_or_remove)
+    protected TextView addOrRemoveText;
+    @BindView(R.id.loadingPanel)
+    protected RelativeLayout progressBar;
+
+    private Storage storage = Storage.getStorage();
+    private Bundle bundle;
+    private Hit photo;
+    private View view;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        view = inflater.inflate(R.layout.item_details, container, false);
+        ButterKnife.bind(this, view);
+        bundle = getArguments();
+        photo = (Hit) bundle.getSerializable("item");
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        showItem(view);
+        ActionBar mActionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        if (mActionBar != null) {
+            mActionBar.show();
+        }
+    }
+
+    private void showItem(View view){
+        if(photo != null) {
+            Picasso.get().load(photo.getLargeImageURL())
+                    .into(imageDetail, new com.squareup.picasso.Callback() {
+
+                        @Override
+                        public void onSuccess() {
+                            progressBar.setVisibility(View.GONE);
+                        }
+
+                        @Override
+                        public void onError(Exception e) {
+                            Log.e("picasso", "Failed to load image");
+                        }
+                    });
+            author.setText(photo.getUser());
+            tags.setText(photo.getTags());
+            views.setText(Objects.toString(photo.getViews()));
+            downloads.setText(Objects.toString(photo.getDownloads()));
+            favourites.setText(Objects.toString(photo.getFavorites()));
+            comments.setText(Objects.toString(photo.getComments()));
+            if(storage.isFavourite(photo, view)){
+                favButton.setImageResource(R.drawable.ic_fav_color);
+                addOrRemoveText.setText(R.string.remove_from_fav);
+            } else{
+                favButton.setImageResource(R.drawable.ic_fav_borders);
+                addOrRemoveText.setText(R.string.add_to_fav);
+            }
+        }
+    }
+
+    @OnClick(R.id.image_detail)
+    void openImage(View v) {
+        MainActivity mainActivity = (MainActivity) v.getContext();
+        FragmentTransaction ft = mainActivity.getSupportFragmentManager()
+                .beginTransaction();
+        ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+        FullScreen fullScreen = new FullScreen();
+        bundle.putSerializable("link", photo.getLargeImageURL());
+        fullScreen.setArguments(bundle);
+        ft.replace(R.id.main_container, fullScreen).addToBackStack(null).commit();
```

Comment:
Again, you have a method inside your Activity for performing this kind of action. Use it instead